protobuf/definition/execution.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+
+package definition;
+option go_package = "github.com/mesg-foundation/core/protobuf/definition";
+
+// Status represents the status of a single execution.
+// Note that a valid execution must have only one status
+// flag at time.
+enum Status {
+  // Unknown status represents any status unknown to execution.
+  Unknown = 0x00;
+
+  // Created is an initial status after execution creation.
+  Created = 0x01;
+
+  // InProgress informs that processing of execution has been started.
+  InProgress = 0x02;
+
+  // Completed is a success status after execution was processed.
+  Completed = 0x04;
+
+  // Failed is an error status after execution was processed.
+  Failed = 0x08;
+}
+
+// Execution represents a single execution run in engine.
+message Execution {
+  // Hash is a unique hash that represents given execution.
+  bytes hash = 1;
+
+  // parentHash is unique hash that represents that triggerd current execution.
+  bytes parentHash = 2;
+
+  // eventID is unique event id.
+  bytes eventID = 3;
+
+  // Status is the current status of execution.
+  Status status = 4;
+
+  // serviceHash is hash of service that can proceed an execution
+  bytes serviceHash = 5;
+
+  // taskKey is a task key from a service definition.
+  string taskKey = 6;
+
+  // inputs is an inptut of execution.
+  bytes inputs = 7;
+
+  // outputs is an output of successful execution.
+  bytes outputs = 8;
+
+  // error is an output of a failed execution.
+  bytes error = 9;
```

Comment:
why bytes for error? It should be a string as we don't have any define structure of what an error is.

Reply:
bytes is for outputs (which is output for success) so it makes sens to make error as bytes (which is output for failed)

Reply:
error is not related to outputs anymore. It related to the state failed.
Error should be a string otherwise it's prone to JSON encode, or protobuf encode, or even dump the GO error struct into this field and NO client will be able to decode it.